spokestack/asr/speech_recognizer.py

```diff
@@ -80,7 +80,16 @@ def _receive(self, context):
             hypothesis = hypotheses[0]
             context.transcript = hypothesis["transcript"]
             context.confidence = hypothesis["confidence"]
-        self._is_final = self._client.response.get("final")
+            if context.transcript:
+                context.event("partial_recognize")
+            else:
+                context.event("timeout")
```

Comment:
timeout events should only be raised if we are actually timing out (we have received an empty final transcript), so you don't want to do it for intermediate responses